docs(firebase_crashlytics): update setup guide (#3860)
FirebaseExtended/flutterfire: docs/crashlytics/usage.mdx

sidebar_label: Usage
---

## Getting started

Enable Firebase Crashytlics in the Firebase console.

> ![Crashytlics - Enable in the console](enable-firebase-crashlytics.png)

Reviewer: I tried running this locally and got this error. I guess the path needs updating.
<img width="946" alt="Screenshot 2021-03-22 at 17 47 28" src="https://user-images.githubusercontent.com/16018629/112034716-b076c280-8b36-11eb-8e37-899819c42d95.png">

> ![Crashytlics - Enable in the console](enable-firebase-crashlytics.png)

Reviewer: ```suggestion
<Image
  src="enable-firebase-crashlytics.png"
  alt="Enable in the console"
/>
```

Until an error has been reported, you will see this screen.

> ![Crashytlics init screen](crashlytics-init.png)

Reviewer: ```suggestion
<Image
  src="crashlytics-init.png"
  alt="Crashytlics init screen"
/>
```

To start using Firebase Crashlytics within your project, import it at the top of your project files:

```dart
import 'package:firebase_crashlytics/firebase_crashlytics.dart';
```

Run an example, such as...

```dart
// Initialize Firebase.
await Firebase.initializeApp();

// Elsewhere in your code
FirebaseCrashlytics.instance.crash();
```

> This will crash the currently running application. You will then need to manually re-open your application on your emulator for Crashlytics to submit the crash report to the Firebase Console.

Reviewer: ```suggestion
This will crash the currently running application. You will then need to manually re-run your application on your emulator for Crashlytics to submit the crash report to the Firebase Console.
```

Restarting the application will then update the Firebase console:

> ![Crashytlics ready](crashlytics-ready.png)

Reviewer: ```suggestion
<Image
  src="crashlytics-ready.png"
  alt="Crashytlics ready"
/>
```

sidebar_label: Usage
---

## Getting started

> Enable Forebase Crashytlics in the Firebase console.

Reviewer 2: Firebase*

![Crashytlics - Enable in the console](enable-firebase-crashlytics.png)

Until an error has been reported, you will see this screen.

![Crashytlics init screen](crashlytics-init.png)

To start using Firebase Crashlytics within your project, import it at the top of your project files:

```dart
import 'package:firebase_crashlytics/firebase_crashlytics.dart';
```

Run an example, such as...

```dart

> // Init application

Reviewer 3: ```suggestion
// Initialize Firebase.
```

> await Firebase.initializeApp();

Reviewer 2: I'd remove this to be honest, it's covered in the getting started guide.

> //Elsehwere in your code

Reviewer 2: Spaces after `//` and `Elsehwere` typo.

FirebaseCrashlytics.instance.crash();
```

> This will restart the application. If using an emulator, please re-run...

Reviewer 2: `please re-run...` -> `please re-run:`

> ```dart

Reviewer 2: bash*

  flutter run
```

> `or restart the app manually through the emulator.`

Reviewer 2: Remove this bit, not needed?

Author: This is a funny one. At times I needed to run one or the other. I kept it in for now as a backup, in case `flutter run` failed.

Open to ideas though!

Reviewer 3: You do have to re-open the app after a crash() call as it only submits on app startup, calling run again re-installs the apk afaik and so loses previous crash data?

Reviewer: When I was firing errors yesterday, I had to crash() and restart app using `flutter run` for reports to send. At least, that was the method I was using for it to work consistently.

> This will restart the application. If using an emulator, please re-run:
> 
> ```bash
>   flutter run
> ```
> 
> `or restart the app manually through the emulator.`

Reviewer 3: ```suggestion
This will crash the currently running application. You will then need to manually re-open your application on your emulator for Crashlytics to submit the crash report to the Firebase Console.
```

> Restarting the application will then update the Firebase console

Reviewer 2: `update the Firebase console` -> `update the Firebase console:`
